doc/rst/tools/protoc-gen-chpl/protoc-gen-chpl.rst

```diff
@@ -0,0 +1,198 @@
+details on ``.proto`` files see the links at the end of this document.
+
+The Chapel implementation only supports the `proto3`_ syntax of protobuf. The
+``.proto`` file starts with an optional package declaration, which helps to prevent
+naming conflicts between different projects.
+
+.. code-block:: proto
+
+  syntax = "proto3";
+  package addressbook;
+
+In Chapel the generated records will be placed in a module matching the ``package``
+name. If the ``package`` name is not specified the module takes the name of the
+proto file with all non-aplhanumeric characters replaced by an ``underscore``.
+
+Next, you have to define your messages. A message is just an aggregate containing
+a set of typed fields. Many standard simple data types are available as field types,
+including int64, int32, float, double, and string.
+
+.. code-block:: proto
+
+  enum Contact {
+    FAMILY = 0;
+    FRIENDS = 1;
+    WORK = 2;
+  }
+
+  message Person {
+    string name = 1;
+    int32 id = 2;  // Unique ID number for this person.
+    string email = 3;
+
+    Contact contact = 4;
+
+    repeated string number = 5; // Person might have more than one number.
+  }
+  
+In the above example the ``Person`` message contains ``string`` type ``name``
+field with field number ``1``, ``int32`` type ``id`` field with field number
+``2`` as well as ``string`` type ``email`` field with field number ``3``. You
+can also define enum types if you want one of your fields to have one of a
+predefined list of values - here you want to specify that a contact type can be
+one of FAMILY, FRIENDS, or WORK. The ``Contact`` type ``contact`` with field number
+``4`` is an example. If a field is repeated, the field may be repeated any number
+of times (including zero). For example, the ``string`` type ``number`` field with
+field number ``4``. The order of the repeated values will be preserved in the protocol
+buffer. If a field is not set, a default value is assigned to the field by Chapel.
+
+
+Compiling your protocol buffers
+===============================
+
+The code generator is integrated with the protoc compiler toolchain
+included in the default Protocol Buffers distribution. Use the ``protoc`` command
+with the ``--chpl_out`` flag to invoke the Chapel code generator and write the
+output ``chpl`` file to a specific location.
+
+.. code-block:: console
+
+  protoc --chpl_out=$DST_DIR $SRC_DIR/addressbook.proto
+
+This generates ``addressbook.chpl`` in your specified directory.
+
+
+The generated file
+==================
+
+The generated ``addressbook.chpl`` file will contain:
+
+* A wrapper module with the name ``addressbook``.
+* A record with the name ``Person``.
+* An enum with the name ``Contact``.
+* ``name_``, ``id_``, ``email_``, ``contact_`` and ``number_`` fields.
+* ``writeToOutputFile`` and ``parseFromInputFile`` functions for serialization/parsing.
+
+You can import this module to a ``chpl`` file and can create an instance of ``Person``
+for populating data.
```

Comment:
I'd use a semi-colon to end this sentence, to more obviously indicate that the code following it is related